[TAU package] Update for rocprofiler-sdk

jordialcaraz opened the pull request:
@spackbot fix style

jordialcaraz committed:
[TAU package] Update for rocprofiler-sdk

jordialcaraz commented:
Do not approve at this moment. The rocprofiler-sdk that spack installs at this moment is not fully working (or at least when I installed it):
https://github.com/spack/spack/issues/49737

jordialcaraz committed:
[@spackbot] updating style on behalf of jordialcaraz

jordialcaraz committed:
Update package.py

jordialcaraz commented:
Pipeline failed due to:
_==> Error: Failed to install rocm-core due to InstallError: No binary found when cache-only was specified
==> Error: Failed to install roctracer-dev-api due to InstallError: No binary found when cache-only was specified_

eugeneswalker commented:
> Pipeline failed due to: _==> Error: Failed to install rocm-core due to InstallError: No binary found when cache-only was specified ==> Error: Failed to install roctracer-dev-api due to InstallError: No binary found when cache-only was specified_

This type of error is happening on many pipelines on other PRs too. It’s a systemic issue unrelated to this PR. It is being discussed internally on CI slack channel. I will post back here when resolution is identified.

eugeneswalker commented:
Any more review necessary before merge? Happy to merge now.

khuck reviewed (commented):
should there be conflicts specified in here?
```
conflicts("+roctracer", when="+rocprofiler-sdk")
```
and so on

khuck commented:
> should there be conflicts specified in here?
> 
> ```
> conflicts("+roctracer", when="+rocprofiler-sdk")
> ```
> 
> and so on

oh, and only with a TAU release newer than X?

jordialcaraz commented:
> should there be conflicts specified in here?
> ```
> conflicts("+roctracer", when="+rocprofiler-sdk")
> ```
> and so on

That is true, will fix when rocprofiler-sdk with spack works, as I may need to do more changes.

jordialcaraz commented:
> > should there be conflicts specified in here?
> > 
> > ```
> > conflicts("+roctracer", when="+rocprofiler-sdk")
> > ```
> > 
> > and so on
> 
> oh, and only with a TAU release newer than X?

Line 111, I wrote that version to use master, will have to change that when we have a new release.

jordialcaraz committed:
Updated conflicts

Wrote the conflicts for rocprofiler-sdk and tried to make the conflict list more clear.

jordialcaraz commented:
@spackbot fix style

jordialcaraz committed:
[@spackbot] updating style on behalf of jordialcaraz
